src/symmetric_memory.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include "mococrw/symmetric_memory.h"
+
+#include <queue>
+#include <cassert>
+
+namespace mococrw
+{
+void QueueOfVectorsMemoryStrategy::write(const std::vector<uint8_t> &&chunk)
+{
+    _queue.push_back(chunk);
```

Comment:
This will cause the copy after all :(
Needs to be `_queue.emplace_back(std::move(chunk))`

Reply:
Is it really needed to move the r-reference here?